Adjust level of verbosity of debug dumps in graph executor T74227880

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43682 Adjust level of verbosity of debug dumps in graph executor T74227880**

Differential Revision: [D23397980](https://our.internmc.facebook.com/intern/diff/D23397980)

Commit:
changed bulk of dump logging into debug logging

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/runtime/graph_executor.cpp:
```diff
@@ -585,38 +585,49 @@ struct GraphExecutorImpl : public GraphExecutorImplBase {
-    GRAPH_DUMP("After LowerGradOf, before specializeAutogradZero", opt_graph);
+    GRAPH_DEBUG(
+        "After LowerGradOf, before specializeAutogradZero\n",
+        opt_graph->toString());
```
Does it work to use `*opt_graph` here instead of `opt_graph->toString()`?  It's maybe slightly cleaner that way.

Lilyjjo replied:
I agree, that does look much cleaner and it also works! I'll make the changes

bertmaher commented on torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:
```diff
@@ -374,9 +444,10 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(
     size_t remaining_bailout_depth) {
   std::lock_guard<std::mutex> lock(compile_mutex);
-  GRAPH_DEBUG("Running ProfilingGraphExecutorImpl ", this);
+  GRAPH_DUMP("Running ProfilingGraphExecutorImpl", graph);
```
I'd probably leave this one as GRAPH_DEBUG; since (I think) it runs on every invocation of the graph, it could dump a ton of output.  Actually, @ZolotukhinM, is this even useful?  I remember seeing a bunch of `Running ProfilingGraphExecutorImpl 0xba5eba11acc01ade` and wondering who would possibly care about the PE pointer.  Do we print it elsewhere, such that it's useful for "linking up" an execution trace?

ZolotukhinM replied:
I think we intentionally printed the pointer here so that we could track how executors are being created during the execution (an executor in some cases could create a new executor, which could create another one, and so forth). Probably @Krovatkin could tell more about the motivation here.

For now I'd recommend leaving this place as it is and revisit it later if we want.

bertmaher reviewed (commented):
Minor comments but looks great to me.  I'll let @ZolotukhinM approve though since he knows this flow better than I.

ZolotukhinM reviewed (approved):
Awesome! Looks good to me as well!

Commit:
Update on "Adjust level of verbosity of debug dumps in graph executor T74227880"




[ghstack-poisoned]

Krovatkin commented on torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp:
```diff
@@ -412,6 +424,7 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(
   ProfilingRecord::removeProfileCounter(copy->block());
   runProfilingOptimizations(copy);
   // cache
+  GRAPH_DUMP("Optimized Graph: ", copy);
```
you added it back!!!!!!!!!!!!!!!! thank you!!!! ))))))))) I was missing this one sorely, since this is what I was grepping for

Krovatkin reviewed (approved):
:shipit:

Commit:
Update on "Adjust level of verbosity of debug dumps in graph executor T74227880"


Differential Revision: [D23397980](https://our.internmc.facebook.com/intern/diff/D23397980)

[ghstack-poisoned]

Commit:
Update on "Adjust level of verbosity of debug dumps in graph executor T74227880"


Differential Revision: [D23397980](https://our.internmc.facebook.com/intern/diff/D23397980)

[ghstack-poisoned]
